editors/vscode/src/client.ts

```diff
@@ -0,0 +1,218 @@
+
+        if (client) {
+            client.stop();
+        }
+
+        client = new LanguageClient(
+            'LambdaPi-VSCode',
+            'Lambdapi Language Server',
+            serverOptions,
+            clientOptions
+        );
+
+        client.onReady().then(() => {
+
+            // Create and show panel
+            const panel = window.createWebviewPanel(
+                'goals',
+                'Goals',
+                ViewColumn.Two,
+                {}
+            );
+
+            window.onDidChangeActiveTextEditor(e => refresh(panel, e));
+            window.onDidChangeTextEditorSelection(e => refresh(panel, e.textEditor));
+
+        });
+        context.subscriptions.push(client.start());
+    };
+
+    commands.registerCommand('extension.vscode-lp.restart', restart);
+
+    restart();
+}
+
+function refresh(panel : WebviewPanel, editor : TextEditor | undefined) {
+
+    if(editor)
+        sendGoalsRequest(editor.selection.active, panel, editor.document.uri);
+
+}
+
+// returns the HTML code of goals environment
+function getGoalsEnvContent(goals : Goal[]){
+
+    let codeHyps : String = ""; //hypothesis HTML code
+    let codeGoals : String = ""; //goals HTML code
+    let codeEnvGoals : String = ""; //result code HTML
+
+    for(let i=0; i < goals.length; i++) {
+
+        codeHyps = `<div class="hypothesis">`;
+        codeGoals = `<div class="pp_goals">`;
+
+        for(let j=0; j<goals[i].hyps.length; j++){
+
+            let hnameCode = `<label class="hname">`
+                + goals[i].hyps[j].hname
+                + `</label>`;
+
+            let htypeCode = `<span class="htype">`
+                + goals[i].hyps[j].htype
+               + `</span> <br/>`;
+
+            codeHyps = codeHyps + hnameCode
+                + `<label class="sep"> : </label>`
+                + htypeCode;
+        }
+
+        let numGoalcode = `<label class="numGoal">`
+            + i + `</label>`;
+
+        let typeGoal = `<span class="goal">`
+            + goals[i].type + `</span>`;
+
+        codeGoals = codeGoals + numGoalcode
+            + `<label class ="sep"> : { </label> `
+            + typeGoal + `<label class ="sep">}</label><br/><br/></div>`;
+
+        codeHyps = codeHyps + `</div>`;
+
+        let codeSep = `<hr/>`;
+        codeEnvGoals = codeEnvGoals + "" + codeHyps + codeSep + codeGoals;
+    }
+
+    // if there is no goals
+    if(goals.length == 0){
+        codeEnvGoals = codeEnvGoals + `No goals`;
+    }
+    return codeEnvGoals;
+}
+
+// Returns the HTML code of the panel and the inset ccontent
+function buildGoalsContent(goals : Goal[]) {
+    
+    let header, footer : String;
+
+    // get the HTML code of goals environment
+    let codeEnvGoals : String = getGoalsEnvContent(goals);
+
+    // #FA8072
+    header =  `<!DOCTYPE html>
+￼       <html lang="en">
+￼       <head>
+￼               <meta charset="UTF-8">
+￼               <link rel="stylesheet" type="text/css" href="style/style.css" >
```

Comment:
There seems to be no file style/style.css currently...

Reply:
Umm maybe it was lost? Let me see.

Reply:
In fact it never worked I think, so this is a todo, we didn't have time to figure out where it should be placed. Added a comment.